kernel/kernel-api/src/test/scala/io/delta/kernel/internal/DeltaLogActionUtilsSuite.scala

```diff
@@ -282,7 +282,7 @@ class DeltaLogActionUtilsSuite extends AnyFunSuite with MockFileSystemClientUtil
     ).toInMemoryList.asScala
 
     assert(commitFiles.forall(fs => FileNames.isCommitFile(fs.getPath)))
-    assert(extractVersions(commitFiles) == Seq(10, 11, 12, 13, 14, 15, 16, 17))
+    assert(extractVersions(commitFiles.toSeq) == Seq(10, 11, 12, 13, 14, 15, 16, 17))
```

Comment:
same comment as above.. how was this compiling with Scala 2.13 before. why do we need to this toSeq change everywhere.

Reply:
does not answer my question.. how was it working before. 
current master is compiling with Scala 2.13 and passing tests isnt it? 
then something in the PR changed something in the 2.13 compilation that changed things for 2.13. the goal of this PR is to just drop 2.12 without changing anything for 2.13.

cc @allisonport-db @scottsand-db

Reply:
@tdas since Kernel is a java-only project for src code we only compile/test it with 2.12 by default. We had to make these exact same sorts of fixes in the 4.0.x release branches.